app/javascript/components/DownloadButton.js

```diff
@@ -115,6 +145,59 @@ function DownloadCommandContainer(props) {
   )
 }
 
+/**
+ * Format number in bytes, with human-friendly units
+ *
+ * Derived from https://gist.github.com/lanqy/5193417#gistcomment-2663632
+ */
+function bytesToSize(bytes) {
+  const sizes = ['bytes', 'KB', 'MB', 'GB', 'TB']
+  if (bytes === 0) return 'n/a'
+
+  // eweitz: Most implementations use log(1024), but such units are
+  // binary and have values like MiB (mebibyte)
+  const i = parseInt(Math.floor(Math.log(bytes) / Math.log(1000)), 10)
+
+  if (i === 0) return `${bytes} ${sizes[i]}`
+  return `${(bytes / (1000 ** i)).toFixed(1)} ${sizes[i]}`
+}
+
+/**
+ * Get introductory text for download command, including download size
+ *
+ * @param {Object} downloadSize Size in bytes, by file type
+ */
+function getLeadText(downloadSize) {
+  // Sum sizes by file type to total size for download
+  const totalSize =
+    Object.values(downloadSize).reduce((prevSize, sizeObj) => {
+      return sizeObj.total_bytes + prevSize
+    }, 0)
+  const prettyBytes = bytesToSize(totalSize)
+
+  // Get file type summary
+  // E.g. "metadata and expression" or
+  // (possibly later) "metadata, cluster, and expression"
+  const ft = Object.keys(downloadSize)
+  let fileTypes = ft.join(' and ')
+  if (ft.length > 2) {
+    const allButLast = ft.slice(0, -1)
+    const last = ft.slice(-1)[0]
+    fileTypes = `${allButLast.join(', ')}, and ${last}`
+  }
+
+  return (`
+    To download ${prettyBytes} in ${fileTypes} files
+    matching your search, copy this command and paste it into your terminal:
+  `)
+}
+
+/** Determine if search has any parameters, i.e. terms or filters */
+function hasSearchParams(params) {
```

Comment:
this seems like a good helper method to move into StudySearchProvider.js, either directly on the context, or just as an exported plain fuction